### DISCUSSION TITLE
additionalMessage / testName in error report

### DISCUSSION THREAD

**Discussion by mtrakal** [LemonAppDev/konsist#1479] [Report a Problem]
### Tell us what version of Konsist you use.

0.16.1

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

We are writing tests in Kotest (Android project, JUnit 5 Jupiter), and using `testName` and `additionalMessage` for tests.

But error output didn't contains them when `exceptionFormat = TestExceptionFormat.SHORT`. In that case we need to use `exceptionFormat = TestExceptionFormat.FULL` to see on which message / testName it fails.

```kotlin
@ExtendWith(Extension::class)
class UseCasesNameNotAllowed : BehaviorSpec() {
    init {
        Given("All classes and interfaces in production code") {
            When("There is a UseCases") {
                Then("It should not exists in codebase") {
                    useCasesInterfaces.assertEmpty(
                        testName = "UseCases is wrong naming convention. Use UseCase instead",
                        additionalMessage = "Failing on `$useCasesInterfaces`",
                    )
                }
            }
        }
    }
}
```

### Actual behaviour
```
> Task :konsist:testDevDebugUnitTest

konsist.rules.usecase.UseCasesNameNotAllowed > Given: All classes and interfaces in production code > When: There is a UseCases > konsist.rules.usecase.UseCasesNameNotAllowed.Then: It should not exists in codebase FAILED
    com.lemonappdev.konsist.core.exception.KoAssertionFailedException at UseCasesNameNotAllowed.kt:18
        Caused by: com.lemonappdev.konsist.core.exception.KoAssertionFailedException at UseCasesNameNotAllowed.kt:18

7 tests completed, 1 failed

> Task :konsist:testDevDebugUnitTest FAILED

FAILURE: Build failed with an exception.
```

### Expected behaviour
```
> Task :konsist:testDevDebugUnitTest

konsist.rules.usecase.UseCasesNameNotAllowed > Given: All classes and interfaces in production code > When: There is a UseCases > konsist.rules.usecase.UseCasesNameNotAllowed.Then: It should not exists in codebase FAILED
    com.lemonappdev.konsist.core.exception.KoAssertionFailedException: Assert 'UseCases is wrong naming convention. Use UseCase instead' failed.
    Failing on `[XUseCases]`
    Declaration list is not empty. It contains values:
    XUseCases.

7 tests completed, 1 failed

> Task :konsist:testDevDebugUnitTest FAILED

FAILURE: Build failed with an exception.
```

In case, that we uses `exceptionFormat = TestExceptionFormat.FULL` log is full of stacktrace which is not needed for fix the issue. It's hard to find issues in that case. So Short log should be enough, but not without message what is wrong in output.

**Comment by igorwojda** [comment]
Unfortunately there is no easy way to get the test name inside Konsist code from KoTest, so test name has to be explicitly passed.
See https://docs.konsist.lemonappdev.com/features/kotest-support

**Reply by mtrakal** [reply]
It's primarily not about testName, but about additional info / test error

```
Failing on `[XUseCases]`
    Declaration list is not empty. It contains values:
    XUseCases.
```

We don't need testName in output, but what is failing and why with 2 options:
- propagate additionalMessage from test `additionalMessage = "Failing on $useCasesInterfaces"` to `TestExceptionFormat.SHORT` output, not only to `TestExceptionFormat.FULL` output with stack trace.
- provide standard error / fail part (which is part of FULL output) in short output:
```
    Declaration list is not empty. It contains values:
    XUseCases.
```
This part is printed by default when TestExceptionFormat.FULL is used. But missing in TestExceptionFormat.SHORT, but it's most important thing to fix issue and is missing in standard output.

**Comment by igorwojda** [comment]
Interesting. We will take a look into this.
For now you can consider usage of dynamic tests https://docs.konsist.lemonappdev.com/advanced/dynamic-konsist-tests
